app/assets/javascripts/discourse/app/mixins/password-validation.js

```diff
@@ -33,18 +33,21 @@ export default Mixin.create({
     "rejectedPasswords.[]",
     "accountUsername",
     "accountEmail",
-    "passwordMinLength"
+    "passwordMinLength",
+    "forceValidationReason"
```

Comment:
Just from reading it, I don't have much idea what this is doing, can we thing of a better name ?

Reply:
I had to introduce this because of:

https://github.com/discourse/discourse/blob/0bdda07e336922f957355aa310225623dea8d3a1/app/assets/javascripts/discourse/app/controllers/create-account.js#L163

Some fields should fail silently and not display the reason message under certain conditions. For example, if the password field is empty, we do not want to show the "please enter a password" message straight away, only on submit. So I made this property when submitting the create account form to change the validation behaviour and show the reason always:

https://github.com/discourse/discourse/blob/0bdda07e336922f957355aa310225623dea8d3a1/app/assets/javascripts/discourse/app/controllers/create-account.js#L435

Reply:
Yes I see.. It's probably related to the fact we use observers in accounts and it creates this kind of issues indeed. I doubt you can do better without heavy refactoring.